ipaclient/install/client.py

```diff
@@ -2345,6 +2345,72 @@ def update_ipa_nssdb():
                                    (nickname, sys_db.secdir, e))
 
 
+def sync_time(options, fstore, statestore):
+    # We assume that NTP servers are discoverable through SRV records
+    # in the DNS.
+    # If that fails, we try to sync directly with IPA server,
+    # assuming it runs NTP
+
+    # disable other time&date services first
+    if options.force_chrony:
+        ntpconf.force_chrony(statestore)
+
+    print("Synchronizing time")
+    print("  [1/1]: Configuring chrony client")
+    logger.info('Synchronizing time with KDC...')
+
+    if not options.ntp_servers:
+        ds = ipadiscovery.IPADiscovery()
+        ntp_servers = ds.ipadns_search_srv(cli_domain, '_ntp._udp',
+                                           None, break_on_first=False)
+    else:
+        ntp_servers = options.ntp_servers
+
+    if ntp_servers:
+        synced_time = ntpconf.configure_chrony(ntp_servers, options.ntp_pool,
+                                               fstore, statestore)
+    else:
+        synced_time = False
+        logger.warning("No SRV records of NTP servers found nor NTP server  "
+                       "address was privided. Skipping chrony configuration")
+
+    if not synced_time:
+        print("Warning: IPA Server was unable to sync time with chrony!")
+        print("         Time synchronization is required "
+              "for IPA Server to work correctly")
+        logger.warning(
+            "Unable to sync time with chrony server, assuming the time "
+            "is in sync. Please check that 123 UDP port is opened, "
+            "and any time server is on network.")
+
+
+def restore_time_sync(statestore, fstore):
+    chrony_configured = statestore.has_state('ntp')
```

Comment:
I was hoping that in general NTP as Network Time Protocol would be acceptable to use as it was because chrony and ntpd are only implementations of this protocol.